xdsl/transforms/csl_stencil_bufferize.py

```diff
@@ -0,0 +1,313 @@
+from collections.abc import Sequence
+from dataclasses import dataclass
+
+from xdsl.context import MLContext
+from xdsl.dialects import arith, bufferization, func, memref, stencil, tensor
+from xdsl.dialects.builtin import (
+    DenseArrayBase,
+    DenseIntOrFPElementsAttr,
+    FunctionType,
+    ModuleOp,
+    TensorType,
+    i64,
+)
+from xdsl.dialects.csl import csl_stencil
+from xdsl.ir import Attribute, Block, BlockArgument, Operation, Region, SSAValue
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    TypeConversionPattern,
+    attr_type_rewrite_pattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.rewriter import InsertPoint
+from xdsl.utils.hints import isa
+from xdsl.utils.isa import isattr
+
+
+def tensor_to_memref_type(t: TensorType[Attribute]) -> memref.MemRefType[Attribute]:
+    """Type conversion from tensor to memref."""
+    return memref.MemRefType(t.get_element_type(), t.get_shape())
+
+
+def to_memref_op(op: SSAValue) -> bufferization.ToMemrefOp:
+    """Creates a `bufferization.to_memref` operation."""
+    assert isa(op.type, TensorType[Attribute])
```

Comment:
nit: Shaped & Container[Attribute]?

Reply:
nvm, it's expected to be a memref by the op, not only for the type

Reply:
usually ok, but the op takes a `TensorType[Attribute]` as input anyway, so might be fine?